src/pricing/balancer.ts

```diff
@@ -0,0 +1,77 @@
+// pricing for balancer traded tokens and balancer LP tokens
+
+import { Address, BigInt, BigDecimal, log } from '@graphprotocol/graph-ts'
+import { BalancerWeightedPool } from '../../generated/templates/GeyserV1/BalancerWeightedPool'
+import { BalancerVault } from '../../generated/templates/GeyserV1/BalancerVault'
+import { integerToDecimal } from '../util/common'
+import { ZERO_BIG_DECIMAL, STABLECOINS, STABLECOIN_DECIMALS } from '../util/constants'
+import { Token } from '../../generated/schema'
+
+// TODO: Add support for more pools other than Weighted Pools.
+
+export function isBalancerLiquidityToken(address: Address): boolean {
+  let pool = BalancerWeightedPool.bind(address);
+
+  let res = pool.try_getNormalizedWeights();
+  if (res.reverted) {
+    return false;
+  }
+  return true;
+}
+
+export function getBalancerLiquidityTokenPrice(address: Address): BigDecimal {
+  // get contracts
+  let pool = BalancerWeightedPool.bind(address);
+  let vaultId = pool.getVault();
+  let vault = BalancerVault.bind(vaultId);
+  let poolId = pool.getPoolId();
+
+  // get pool stats
+  let poolTokens = vault.getPoolTokens(poolId);
+  let tokenAddresses = poolTokens.value0;
+  let tokenBalances = poolTokens.value1;
+  let weights = pool.getNormalizedWeights();
+  let totalSupply = integerToDecimal(pool.totalSupply());
+
+  // find stable coin
+  let stableCoinPoolIndex: i32 = -1;
+  let foundStableCoin: i32 = -1;
```

Comment:
can we name these more consistently with `idx`, `index`, or `found`?

Reply:
good catch